app/Services/TripsService.php

```diff
@@ -55,14 +57,44 @@ public function create(CreateTripRequest $request, $user): Trip
     }
 
     /**
-     * @param Trip $trip
+     * Update trip service
+     *
+     * @param $tripId
      * @param UpdateTripRequest $request
      * @param $user
      * @return Trip
+     * @throws TripNotFoundException
+     * @throws UserCantEditTripException
      */
-    public function update(Trip $trip, UpdateTripRequest $request, $user): Trip
+    public function update($tripId, UpdateTripRequest $request, $user)
     {
-        return $trip;
+        $trip = $this->tripRepository->findTripById($tripId);
+
+        if(is_null($trip)) {
+            throw new TripNotFoundException("Trip not found");
+        }
+
+        if($user->id != $trip->user_id) {
```

Comment:
Use the same approach like we used https://github.com/BinaryStudioAcademy/bsa-2017-bebeep/blob/d20408e5aac1bde6bbf1edfb0abc1ae82235942f/app/Services/TripsService.php#L103 here to check if user can delete trip